cloud/pkg/cloudhub/channelq/channelq.go

```diff
@@ -33,13 +39,16 @@ type ChannelMessageQueue struct {
 
 	objectSyncLister        reliablesyncslisters.ObjectSyncLister
 	clusterObjectSyncLister reliablesyncslisters.ClusterObjectSyncLister
+
+	crdClient crdClientset.Interface
 }
 
 // NewChannelMessageQueue initializes a new ChannelMessageQueue
 func NewChannelMessageQueue(objectSyncLister reliablesyncslisters.ObjectSyncLister, clusterObjectSyncLister reliablesyncslisters.ClusterObjectSyncLister) *ChannelMessageQueue {
 	return &ChannelMessageQueue{
 		objectSyncLister:        objectSyncLister,
 		clusterObjectSyncLister: clusterObjectSyncLister,
+		crdClient:               client.GetCRDClient(),
```

Comment:
I think it would be better to assign this field as a parameter.

Reply:
Sorry, what do you mean about assigning this field as a parameter?

Reply:
```golang
func NewChannelMessageQueue(objectSyncLister reliablesyncslisters.ObjectSyncLister, clusterObjectSyncLister reliablesyncslisters.ClusterObjectSyncLister, crdClient crdClientset.Interface) *ChannelMessageQueue
```
Like this `crdClient`, and initialize the incoming parameters in the upper layer.